### PR TITLE
Teams

### PR DESCRIPTION
**Pull request by birdglove2**

### CODE REVIEW

File: users/src/routes/__test__/user_test/user-info.test.ts [hunk 1]
```diff
@@ -42,30 +52,46 @@ describe('USER--INFO: Get list of teams from user', () => {
   });
 
   it('should return list of team if the user is found and is a member of one or more team(s)', async () => {
-    const user = await User.create({
+    const user1 = await User.create({
```
Comment (palsp) [inline users/src/routes/__test__/user_test/user-info.test.ts:55]: consider move creation logic ( preparation before test ) to one helper function

File: users/src/routes/team.routes.ts [hunk 2]
```diff
@@ -18,19 +17,11 @@ router.get('/members/:name', requireUser, teamController.getTeamMember);
 router.post('/', requireUser, createTeamValidator, validateRequest, teamController.createTeam);
 
 router.post(
-  '/members',
+  '/invite-member',
```
Comment (palsp) [inline users/src/routes/team.routes.ts:20]: we need to discuss naming route, I think team and connection need to be consistent

File: users/src/routes/user.routes.ts [hunk 3]
```diff
@@ -36,4 +40,12 @@ router.post(
   userController.manageStatus
 );
 
+router.post(
```
Comment (palsp) [inline users/src/routes/user.routes.ts:43]: this should belong to the team route?

File: users/src/controllers/user.controller.ts [hunk 4]
```diff
@@ -149,22 +150,46 @@ export const findRelation = async (req: Request, res: Response) => {
   res.status(200).send(response);
 };
 
+export const requetToJoinTeam = async (req: Request, res: Response) => {
+  const user = req.user!;
+
+  const { teamName } = req.body;
+
+  const team = await Team.findOne({ where: { name: teamName } });
```
Comment (palsp) [inline users/src/controllers/user.controller.ts:158]: consider using include members

File: users/src/models/isMember.model.ts [hunk 5]
```diff
@@ -44,6 +47,13 @@ class IsMember extends Model<IsMemberAttrs, IsMemberCreationAttrs> implements Is
           values: Object.values(TeamStatus),
           allowNull: false,
         },
+        sender: {
+          // TODO add this enum to common --- not using boolean because the attrs name could be confused.
```
[the remark this answers is not in the document]
Comment (birdglove2) [reply users/src/models/isMember.model.ts:51]: birdglove2

File: users/src/routes/user.routes.ts (hunk 3 above)
Comment (birdglove2) [reply users/src/routes/user.routes.ts:43]: there's two routes of manageStatus
one for user: accepting invitation
one for team: accepting request

File: users/src/models/isMember.model.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (palsp) [reply users/src/models/isMember.model.ts:51]: add la

File: users/src/routes/user.routes.ts (hunk 3 above)
Comment (palsp) [reply users/src/routes/user.routes.ts:43]: for the user, it still related to teams in my opinion

File: users/src/models/index.ts [hunk 6]
```diff
@@ -100,7 +103,7 @@ export const autoMigrate = (sequelize: Sequelize) => {
     foreignKey: 'teamName',
   });
 
-  User.belongsToMany(Team, { as: 'member', through: isMember, foreignKey: 'userId' });
+  User.belongsToMany(Team, { as: 'request', through: isMember, foreignKey: 'userId' });
```
[the remark this answers is not in the document]
Comment (birdglove2) [reply users/src/models/index.ts:106]: ok

File: users/src/models/user.model.ts [hunk 7]
```diff
@@ -332,13 +334,48 @@ class User extends Model<UserAttrs, UserCreationAttrs> {
...
+      userId: this.id,
+      status: TeamStatus.Pending,
+      sender: 'user',
+    });
+  }
+
+  // the team(s) that created by myself + the team(s) that I belongs to
+  public async getMyTeams(): Promise<Team[]> {
+    let teams: Team[] = [];
+    const myOwnTeams = await Team.findAll({ where: { creatorId: this.id } });
```
[the remark this answers is not in the document]
Comment (birdglove2) [reply users/src/models/user.model.ts:360]: the creator does not have members relationship as you said earlier

**Pull request merged**